### ISSUE TITLE
PLATFORMIO_HOME_DIR and PLATFORMIO_LIB_EXTRA_DIRS environment variables are not set for upload() command

### ISSUE THREAD

**Issue by ryanfobel** [wheeler-microfluidics/platformio-helpers#3]
# Problem

The environment variables `PLATFORMIO_HOME_DIR` and `PLATFORMIO_LIB_EXTRA_DIRS` are typically set when activating a conda environment (via "%CONDA_PREFIX%\etc\conda\activate.d" scripts); however, the `upload()` method launches a command via `suprocess.check_call()`, and within the environment this command is executed, these variables are not set.

This caused differences in behavior between uploads initiated from within a conda environment and from within MicroDrop. In general, it seems that the environment that MicroDrop is running under may not be fully **activated** which may lead to other unexpected behavior in the future.

# Workaround

As a temporary workaround, we can manually set these environment variables within the `upload()` method; however, this doesn't seem like an ideal solution.

```python
import conda_helpers as ch
env['PLATFORMIO_HOME_DIR'] = str(ch.conda_prefix() /
    'share' / 'platformio')
env['PLATFORMIO_LIB_EXTRA_DIRS']= str(ch.conda_prefix() /
    'Library' / 'include' / 'Arduino')
```

**Comment by cfobel** [comment]
As a slightly more general solution, this can be solved by running the upload command in an activated Conda environment.  This can be accomplished by including the activation as part of the command executed by the `subprocess.check_call` function, i.e.,:

    call <conda prefix>\Scripts\activate.bat & <command>

For example, see the [`conda_exec` function][1] in the [`conda-helpers` package][2].

# Long-term solution

This should likely be fixed by the calling process.  If the calling process is executing within an **activated** Conda environment, the `PLATFORMIO...` environment variables would already be set.

For the case of executing `upload` from MicroDrop, see wheeler-microfluidics/microdrop-launcher#10.

[1]: https://github.com/wheeler-microfluidics/conda-helpers/blob/ac3eed5ef083de6a387dbb9a0d916a7854c67632/conda_helpers/__init__.py#L287-L290
[2]: https://github.com/wheeler-microfluidics/conda-helpers

**Comment by cfobel** [comment]
As of [`v0.7.2`][1] of wheeler-microfluidics/microdrop-launcher, MicroDrop is launched in an **activated** Conda environment.

[1]: https://github.com/wheeler-microfluidics/microdrop-launcher/releases/tag/v0.7.2

**Comment by cfobel** [comment]
Fixed in [release `v0.3.2`][1].

[1]: https://github.com/wheeler-microfluidics/platformio-helpers/releases/tag/v0.3.2
